Mathlib/Algebra/Order/Group/Abs.lean

```diff
@@ -510,6 +522,10 @@ theorem eq_of_abs_sub_nonpos (h : |a - b| ≤ 0) : a = b :=
   eq_of_abs_sub_eq_zero (le_antisymm h (abs_nonneg (a - b)))
 #align eq_of_abs_sub_nonpos eq_of_abs_sub_nonpos
 
+theorem abs_sub_pos_iff_ne : 0 < |a - b| ↔ a ≠ b :=
+  ⟨fun h eq => lt_irrefl (0 : α) (by rwa [eq, sub_self, abs_zero] at h),
+  fun h => lt_iff_not_le.mpr (mt eq_of_abs_sub_nonpos h)⟩
```

Comment:
Perhaps:
```suggestion
@[simp]
theorem abs_sub_nonpos_iff_eq : |a - b| ≤ 0 ↔ a = b :=
  ⟨eq_of_abs_sub_nonpos, by rintro rfl; rw [sub_self, abs_zero]⟩

theorem abs_sub_pos_iff_ne : 0 < |a - b| ↔ a ≠ b :=
  not_le.symm.trans abs_sub_nonpos_iff_eq.not
```

though I might drop the `iff_eq`/`iff_ne` suffixes as well

Reply:
The lhs or `abs_sub_nonpos` simplifies to `a - b = 0`, so I can't really have it as `@[simp]`